### PR TITLE
Feature win share caching mode support

### PR DESCRIPTION
**Pull request by daBONDi**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
module/plugin/windows/win_share

##### ANSIBLE VERSION
```
ansible 2.2.0.0
```

##### SUMMARY
Add Support to Set the Caching Mode for the SMB Share

### CODE REVIEW

**Comment by daBONDi** [comment]
Looks like Shippable is not able to connect to the IRC Server :-)

```
Sending IRC notification  
No connection returned for server: [object Object]
```

File: lib/ansible/modules/windows/win_share.py [hunk 1]
```diff
@@ -84,7 +84,12 @@
       - Specify user list that should get no access, regardless of implied access on share, separated by comma.
     required: no
     default: none
-author: Hans-Joachim Kliemeck (@h0nIg)
+  caching_mode:
+    description:
+      - Set the CachingMode for this share Options: BranchCache, Documents, Manual, None, Programs, Unkown
+    requred: no
+    default: Manual
```
Comment (mattclay) [inline lib/ansible/modules/windows/win_share.py]: CI failure due to the `:` in the description:

```
2017-02-01 18:27:46 ============================================================================
2017-02-01 18:27:46 lib/ansible/modules/windows/win_share.py
2017-02-01 18:27:46 ============================================================================
2017-02-01 18:27:46 ERROR: DOCUMENTATION.options.caching_mode.description.0: expected basestring @ data['options']['caching_mode']['description'][0]. Got None
```

You probably meant this instead?

```yaml
    description:
      - Set the CachingMode for this share.
    choices:
      - BranchCache
      - Documents
      - Manual
      - None
      - Programs
      - Unknown
    required: no
```

Also, note the typos in `Unkown` and `requred` (fixed in my example above).

**Comment by mattclay** [comment]
@daBONDi Please remove the `@` mention from your commit message. That will avoid spurious GH notifications. Thanks.

**Comment by daBONDi** [comment]
Sorry don't know that, i'm a git/hub n00b :)
Removed it

**Comment by mattclay** [comment]
@daBONDi I'm still seeing an `@` mention in commit: c908bbfcc8ce69006cc239afb0763627330940b8

You'll need to amend the commit and force push to overwrite it on GH.

**Comment by daBONDi** [comment]
Will make a new PR, i'm not able to remove the "@" sign, sorry :-(

**Pull request closed without merging**